lib/cli/generators/GATSBY/index.js

```diff
@@ -0,0 +1,55 @@
+import mergeDirs from 'merge-dirs';
+import path from 'path';
+import fs from 'fs';
+import { getVersions, getPackageJson, writePackageJson } from '../../lib/helpers';
+
+export default async npmOptions => {
+  const [
+    storybookVersion,
+    actionsVersion,
+    linksVersion,
+    addonsVersion,
+    babelCoreVersion,
+    babelRuntimeVersion,
+  ] = await getVersions(
+    npmOptions,
+    '@storybook/react',
+    '@storybook/addon-actions',
+    '@storybook/addon-links',
+    '@storybook/addons',
+    'babel-core',
+    'babel-runtime'
+  );
+
+  mergeDirs(path.resolve(__dirname, 'template/'), '.', 'overwrite');
+
+  const packageJson = getPackageJson();
+
+  packageJson.dependencies = packageJson.dependencies || {};
+  packageJson.devDependencies = packageJson.devDependencies || {};
+  packageJson.devDependencies['@storybook/react'] = storybookVersion;
+  packageJson.devDependencies['@storybook/addon-actions'] = actionsVersion;
+  packageJson.devDependencies['@storybook/addon-links'] = linksVersion;
+  packageJson.devDependencies['@storybook/addons'] = addonsVersion;
+
+  // The specific version (7.1.1) of babel-loader solves a problem caused by Gatsby and Storybook using different versions of Webpack.
+  // Details on the error can be and the solution was found at: https://github.com/babel/babel-loader/issues/505#issuecomment-324268931.
+  packageJson.devDependencies['babel-loader'] = '7.1.1';
```

Comment:
This can overwrite the version that user had before running `getstorybook`. Let's at least output a warning in that case